### PR TITLE
Add zero license case to plurals

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This adds the zero case for licenses, since it turns out that some apps actually have this case.

## Tests

- Passes tests.

## Breaking Change

- [X] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter_localizations/lib/src/l10n/generated_material_localizations.dart [hunk 1]
```diff
@@ -449,6 +452,9 @@ class MaterialLocalizationAm extends GlobalMaterialLocalizations {
   @override
   String get licensesPackageDetailTextOther => '\$licenseCount licenses';
 
+  @override
```
Comment (darrenaustin) [inline packages/flutter_localizations/lib/src/l10n/generated_material_localizations.dart:455]: Did you hand modify this file? It is a file that we generate from the *.arb files with the `gen_localizations.dart` script. See comment at the top of `material_localizations.dart' for instructions on how to use it.

**Comment by HansMuller** [comment]
This line in packages/flutter/test/material/localizations_test.dart needs to change

```dart
    expect(() => localizations.licensesPackageDetailText(0), throwsAssertionError);
```

**Comment by HansMuller** [comment]
The method is used in packages/flutter/lib/src/material/about.dart, and it cannot return null:

```dart
        subtitle: Text(MaterialLocalizations.of(context).licensesPackageDetailText(numberLicenses)),
```

File: packages/flutter_localizations/lib/src/l10n/generated_material_localizations.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter_localizations/lib/src/l10n/generated_material_localizations.dart:455]: No, I didn't hand modify this.

**Comment by gspencergoog** [comment]
OK, I think I've got it right now.  PTAL

File: packages/flutter/lib/src/material/material_localizations.dart [hunk 2]
```diff
@@ -912,12 +912,14 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
...
   @override
   String licensesPackageDetailText(int licenseCount) {
-    assert (licenseCount >= 1);
+    assert (licenseCount >= 0);
     switch (licenseCount) {
+      case 0:
+        return 'No licenses.';
       case 1:
-        return '1 license';
+        return '1 license.';
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/material_localizations.dart:920]: Assuming that the additional punctuation makes sense; the translations don't include a period.

**Review by HansMuller — APPROVED** [review]
LGTM

I think we've just rediscovered that we don't have a way to include plural cases (like zero) for only some locales. The new gen_localizations tool for apps does handle that, but the rather specialized code generator and patterns we're using for flutter itself does not.

**Comment by gspencergoog** [comment]
I'm going to just revert the new license page and re-land it with updated logic for zero licenses.

See https://github.com/flutter/flutter/pull/60548 for the revert.

**Pull request closed without merging**